app/brave_main_delegate.cc

```diff
@@ -150,6 +156,12 @@ bool BraveMainDelegate::BasicStartupComplete(int* exit_code) {
   command_line.AppendSwitchASCII(switches::kSyncServiceURL,
                                  "https://no-thanks.invalid");
 
+  // GaiaURL is not gated behind the preference because
+  // GaiaURLs are initialized before profile init, this is okay
+  // because kSigninAllowedOnNextStartup will still be disabled
+  command_line.AppendSwitchASCII(switches::kGaiaUrl,
```

Comment:
why not reverting the patch in `google_apis/gaia/gaia_urls.cc` and then specify `switches::kLsoUrl`